test/integration/targets/azure_rm_functionapp/tasks/main.yml

```diff
@@ -1,14 +1,14 @@
 - name: create storage account for function apps
   azure_rm_storageaccount:
     resource_group: '{{ resource_group }}'
-    name: azfunccistor4
+    name: sa{{ resource_prefix | replace("-","x") }}
```

Comment:
Looks good overall. The only thing I'd change is to move the resource_prefix modification to a set_fact task so you can reference a variable without repeating the string replacement on each usage.